src/core/SpriteAnimator.tsx

```diff
@@ -127,20 +209,144 @@ export const SpriteAnimator: React.FC<SpriteAnimatorProps> = /* @__PURE__ */ Rea
+          } else if (spriteData.current && typeof spriteData.current === 'object' && frameName) {
+            totalFrames.current = spriteData.current.frames[frameName].length
+          } else {
+            totalFrames.current = 0
+          }
+
+          if (playBackwards) {
+            currentFrame.current = totalFrames.current - 1
+          }
+
+          const { w, h } = getFirstFrame(spriteData.current?.frames ?? []).sourceSize
+          const aspect = calculateAspectRatio(w, h)
+
+          setAspect(aspect)
+          if (matRef.current) {
+            matRef.current.map = textureData
+          }
+        }
+
+        // buffer for instanced
+        if (instanceItems) {
+          for (var i = 0; i < instanceItems.length; i++) {
+            const keys = Object.keys(spriteData.current?.frames ?? {})
+            const randomKey = keys[Math.floor(Math.random() * keys.length)]
+
+            frameBuffer.current.push({
+              key: i,
+              frames: spriteData.current?.frames ?? [],
+              selectedFrame: randomKey,
+              offset: { x: 0, y: 0 },
+            })
+          }
+        }
+
+        setSpriteTexture(textureData)
+      },
+      [frameName, getFirstFrame, instanceItems, numberOfFrames, playBackwards]
+    )
+
+    // modify the sprite material after json is parsed and state updated
+    const modifySpritePosition = React.useCallback(() => {
+      if (!spriteData.current) return
+      const {
+        meta: { size: metaInfo },
+        frames,
+      } = spriteData.current
+
+      const { w: frameW, h: frameH } = Array.isArray(frames)
+        ? frames[0].sourceSize
+        : frameName
+        ? frames[frameName]
+          ? frames[frameName][0].sourceSize
+          : { w: 0, h: 0 }
+        : { w: 0, h: 0 }
+
+      if (matRef.current && matRef.current.map) {
+        matRef.current.map.wrapS = matRef.current.map.wrapT = THREE.RepeatWrapping
+        matRef.current.map.center.set(0, 0)
+        matRef.current.map.repeat.set((1 * flipOffset) / (metaInfo.w / frameW), 1 / (metaInfo.h / frameH))
+      }
+      //const framesH = (metaInfo.w - 1) / frameW
+      const framesV = (metaInfo.h - 1) / frameH
+      const frameOffsetY = 1 / framesV
+      if (matRef.current && matRef.current.map) {
+        matRef.current.map.offset.x = 0.0 //-matRef.current.map.repeat.x
+        matRef.current.map.offset.y = 1 - frameOffsetY
+      }
+
+      if (onStart) {
+        onStart({
+          currentFrameName: frameName ?? '',
+          currentFrame: currentFrame.current,
+        })
+      }
+    }, [flipOffset, frameName, onStart])
+
+    // Move these into refs to prevent them from triggering re-renders
+    const loadAssetsRef = React.useRef((imageUrl: string, dataUrl: string) => {
+      loadJsonAndTexture(imageUrl, dataUrl)
+    })
+
+    const parseDataRef = React.useRef((texture: THREE.Texture, data: SpriteData) => {
+      parseSpriteDataLite(texture, data)
+    })
```

Comment:
putting those 2 funcs into refs is really weird to me.

Reply:
I did so because if added as dependencies (even if they are useCallback) they still cause un-intentional re-renders, one solution was to omit them from the dependency array but that won't pass from eslint, this was the other way, perhaps there is another way more common to deal with this scenario?

Reply:
why those 2 funcs would change "unintentional"ly?

Reply:
not sure, I'll have to dig deeper to see which of their own dependencies trigger the change

Reply:
I think https://github.com/pmndrs/drei/pull/2139 should be fixed with this PR, as both are coupled

Reply:
Added type fixing on useSpriteLoader and amended the documentation examples to showcase the replaced properties like scale with meshProps

Reply:
ok but still using your wrapped version of your 2 funcs?

Reply:
not using them anymore, I removed them now (not sure why eslint didn't warn about unused vars)